### ISSUE TITLE
Questions about the experimental environment

### ISSUE THREAD

**Issue by ZhangJinian** [whuyuxia/SMILEcode#1]
During the reproduction experiment, I encountered a mismatch between python packages and I wanted to know what version of python was used for the experiment

**Comment by whuyuxia** [comment]
My python version is 3.6.8, thank you for your question, and I have added this information in my Readme.md. If you have any other problem, feel free to contact me!

**Comment by ZhangJinian** [comment]
Thank you, the code is ready to run. But I ran into some other issues: I trained with the SMILE dataset, and the code only commented out "os.environ['CUDA_VISIBLE_DEVICES']=args.use_gpu", but the loss was all nan. Do you have any ideas?
![image](https://github.com/user-attachments/assets/f2c63fa2-3a87-491c-bb8a-14681742e60e)

**Comment by whuyuxia** [comment]
It seems your training process is running on the CPU, which might indicate that the computer is either not specifying the correct GPU or does not have a GPU. If you still want to proceed with training on the CPU, you can set os.environ['CUDA_VISIBLE_DEVICES'] = "" or use_gpu = '' in the train.sh file to disable GPU usage.

Additionally, you need to install the CPU version of TensorFlow by running:
pip install tensorflow==1.15

This should prevent NaN values during the training process. 
![image](https://github.com/user-attachments/assets/1530f3ac-df51-46a7-b553-976d9a976f61)


Moreover, I highly recommend using a GPU for training. For instance, with a batch size of 4, a single TiTAN GPU can complete an epoch in about 22 minutes, whereas CPU training might take over 14 hours. I can give you the best model I have trained in my paper  if you need it, directly for testing！

**Comment by whuyuxia** [comment]
It seems your training process is running on the CPU, which might indicate that the computer is either not specifying the correct GPU or does not have a GPU. If you still want to proceed with training on the CPU, you can set os.environ['CUDA_VISIBLE_DEVICES'] = "" or use_gpu = '' in the train.sh file to disable GPU usage.

Additionally, you need to install the CPU version of TensorFlow by running:
pip install tensorflow==1.15

This should prevent NaN values during the training process.


Moreover, I highly recommend using a GPU for training. For instance, with a batch size of 4, a single TiTAN GPU can complete an epoch in about 22 minutes, whereas CPU training might take over 14 hours. I can give you the best model I have trained in my paper if you need it, directly for testing！




-----原始邮件-----
发件人:Zhang ***@***.***>
发送时间:2024-10-16 11:43:39 (星期三)
收件人: whuyuxia/SMILEcode ***@***.***>
抄送: whuyuxia ***@***.***>, Comment ***@***.***>
主题: Re: [whuyuxia/SMILEcode] Questions about the experimental environment (Issue #1)






Thank you, the code is ready to run. But I ran into some other issues: I trained with the SMILE dataset, and the code only commented out "os.environ['CUDA_VISIBLE_DEVICES']=args.use_gpu", but the loss was all nan. Do you have any ideas?
image.png (view on web)
